### PR TITLE
Split dockerfiles into major releases for php7 and php8 requirements

### PR DESCRIPTION
**Pull request by othercorey**
@markstory This split the deploy actions, but they need to be updated with the actual server info.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,17 +4,19 @@ ELASTIC_SOURCE_DIR=../elastic-search
...
-.PHONY: build-all
+.PHONY: build-php74
+.PHONY: build-php81
 .PHONY: build-active-and-missing
 .ALL: help
 
 # Versions that can be built.
-CAKEPHP_VERSIONS = 3.0 3.1 3.2 3.3 3.4 3.5 3.6 3.7 3.8 3.9 3.10 4.0 4.1 4.2 4.3 4.next 5.0
+CAKEPHP_PHP74_VERSIONS = 3.0 3.1 3.2 3.3 3.4 3.5 3.6 3.7 3.8 3.9 3.10 4.0 4.1 4.2 4.3 4.next
+CAKEPHP_PHP81_VERSIONS = 5.0
```
Comment (othercorey) [inline Makefile]: We could split this on cake3/4 and cake5. We'll eventually need this split for Chronos. What should the naming convention be? 

The api nginx would probably be easier to maintain with a clean major version split.

File: .github/workflows/deploy_php81.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+name: 'deploy_2x'
```
Comment (markstory) [inline .github/workflows/deploy_php81.yml]: Should we rename this?

File: Dockerfile.php7 [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    php8-intl \
+    php8-json \
+    php8-mbstring \
+    php8-openssl \
+    php8-phar \
+    php8-simplexml \
+    php8-tokenizer \
+    php8-xml \
+    php8-xmlwriter \
+    php8-zip \
```
Comment (markstory) [inline Dockerfile.php7]: Do you need php8 in the php8 container?

File: Makefile (hunk 1 above)
Comment (markstory) [reply Makefile]: Can we group targets by the php versions they need?

Comment (othercorey) [reply Makefile]: Do you mean the `build-cakephp-version` targets? I wasn't sure how to do this. We need to specify a config file based on the major release for cakephp 3 and 4 and eventually 5.

File: Dockerfile.php7 (hunk 3 above)
Comment (othercorey) [reply Dockerfile.php7]: The api docs require php8.

File: Makefile (hunk 1 above)
Comment (markstory) [reply Makefile]: I was thinking more that we would have different dockerfiles for each 'block' of the API and push each docker file to its own container in dokku and use the `site-api` proxy to glue all the sites together again.

The makefile would have `build-2x` and `build-3x` targets that we could use in the dockerfiles.

Comment (othercorey) [reply Makefile]: If you're open to configuring more dokku servers, that would work. The php version split should be along major releases.

Comment (markstory) [reply Makefile]: I don't mind adding more dokku apps. We have a bunch now and the nginx that dokku spawns is pretty lightweight.

**Review by Arhell — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (othercorey) [reply Makefile]: I split up the Dockerfiles based on major releases and created separate deploy configs for each. The deploy configs are all copies of the original still until we have the dokku stuff setup.

Comment (othercorey) [reply Makefile]: Ended up rolling back that change and creating a single, multi-stage Dockerfile with different php builds stages.

Comment (markstory) [reply Makefile]: Looks good. A multistage build is a clever solution.

**Pull request merged**